skimage/draw/_draw.pyx

```diff
@@ -392,12 +383,16 @@ def circle_perimeter(Py_ssize_t cy, Py_ssize_t cx, Py_ssize_t radius,
                 d = d + 2 * (y - x - 1)
                 y = y - 1
                 x = x + 1
+
+    ry = (np.array(rr, dtype=np.float_) + cy)
+    rx = (np.array(cc, dtype=np.float_) + cx)
+    ry = np.round(ry)
+    rx = np.round(rx)
+
     if shape is not None:
-        return _coords_inside_image(np.array(rr, dtype=np.intp) + cy,
-                                    np.array(cc, dtype=np.intp) + cx,
-                                    shape)
-    return (np.array(rr, dtype=np.intp) + cy,
-            np.array(cc, dtype=np.intp) + cx)
+        return _coords_inside_image(ry.astype(int), rx.astype(int), shape)
```

Comment:
I think this is what is causing the test failures in  AppVeyor: both arrays should be converted to `np.intp`, not directly to `int`.